hathor/utils/api.py

```diff
@@ -0,0 +1,54 @@
+#  Copyright 2023 Hathor Labs
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from __future__ import annotations
+
+from typing import List, Optional, TypeVar
+
+from pydantic import Field, ValidationError, validator
+from twisted.web.http import Request
+
+from hathor.api_util import get_args
+from hathor.utils.pydantic import BaseModel
+
+T = TypeVar('T')
+
+
+def _list_to_single_item(v: List[T]) -> Optional[T]:
+    assert len(v) <= 1, 'expected one value at most'
+
+    return None if not len(v) else v[0]
+
+
+class QueryParams(BaseModel):
```

Comment:
Missing docstring. Please add a note about in which cases this class is useful.

Reply:
Done! Also, I've added tests for it.

Reply:
Thanks!